eclair-core/src/main/scala/fr/acinq/eclair/router/RouteCalculation.scala

```diff
@@ -88,7 +92,12 @@ object RouteCalculation {
                 case None => fr.extraEdges.map(GraphEdge(_)).find(e => e.desc.shortChannelId == shortChannelId && e.desc.a == currentNode).map(_.desc)
               }
               channelDesc_opt.flatMap(c => g.getEdge(c)) match {
-                case Some(edge) => (edge.desc.b, previousHops :+ ChannelHop(getEdgeRelayScid(d, localNodeId, edge), edge.desc.a, edge.desc.b, edge.params))
+                case Some(edge) =>
+                  val hop = edge.params match {
+                    case params: ChannelRelayParams => ChannelHop(getEdgeRelayScid(d, localNodeId, edge), edge.desc.a, edge.desc.b, params)
+                    case params: HopRelayParams.FromBlindedRoute => BlindedHop(params.dummyId, params.route, params.paymentInfo)
```

Comment:
Same here, blinded route should only come from BOLT12 invoices, they should not come from the router.

Reply:
But then they are injected via `extraEdges`, which can make sense?